glmodelview.cpp

```diff
@@ -24,31 +30,32 @@ GlModelView::GlModelView(QWidget *parent) :
 
 void GlModelView::initializeGL()
 {
+    /* TODO
     if (!context()->isValid())
     {
         QMessageBox::warning(this,"No OpenGL Driver","Depth Map generation is still available");
-    }
-
+    }*/
 
-    if (QGLShader::hasOpenGLShaders(QGLShader::Vertex,context()) )
+    if (QOpenGLShader::hasOpenGLShaders(QOpenGLShader::Vertex,context()) )
     {
         QString declarations= "//uniform mat4 gl_ModelViewMatrix;  uniform mat4 gl_ProjectionMatrix; attribute vec4 gl_Vertex;\n";
         QString codev=       "void main()  {gl_Position = gl_ModelViewProjectionMatrix * gl_Vertex;"\
                 "float zmax=-1.0;"\
                 "float zmin=1.0;"\
                 "float scale=(zmax-zmin);float offs=-zmin;float z=(gl_Position.z/gl_Position.w); gl_FrontColor = vec4((z+offs)/scale,(z+offs)/scale,(z+offs)/scale,1.0); } ";
-        QGLShader shaderv(QGLShader::Vertex);
+        QOpenGLShader shaderv(QOpenGLShader::Vertex);
         bool compile_success=shaderv.compileSourceCode(declarations+codev);
         if (!compile_success)
         {
             compile_success=shaderv.compileSourceCode(codev);
         }
         if (compile_success)
         {
-            QGLShaderProgram program(context());
-            program.addShader(&shaderv);
-            program.link();
-            program.bind();
+            QOpenGLShaderProgram *program;
+            program = new QOpenGLShaderProgram(this);
```

Comment:
Where is this memory allocation getting released?

Reply:
Good question ;). I'm a bit swamped with work, but I'll try to update the PR.